src/filesystem_helper.cpp

```diff
@@ -341,6 +343,38 @@ path create_temp_directory(const std::string & base_name, const path & parent_pa
   return final_path;
 }
 
+std::filesystem::path create_temporary_directory(
+  const std::string & base_name, const std::filesystem::path & parent_path, size_t max_tries)
+{
+  auto it =
+    std::find(base_name.begin(), base_name.end(), std::filesystem::path::preferred_separator);
+  if (it != base_name.end()) {
```

Comment:
I think we can make this simpler by using the `string.find` method:

```suggestion
  if (base_name.find(std::filesystem::path::preferred_separator) != base_name.npos) {
```

Reply:
Fixed in the https://github.com/ros2/rcpputils/pull/198/commits/8b4087d95df458948f20f0d533c56330d5733606 commit